crates/codegen/parser/runtime/src/napi_interface/cst.rs

```diff
@@ -16,6 +15,16 @@ pub enum NodeType {
     Token,
 }
 
+impl RustNode {
+    /// Converts a Rust node into a choice of NAPI-exposed wrapper structs.
+    pub fn into_either(self) -> Either<RuleNode, TokenNode> {
```

Comment:
`Either<A,B>` is type-gen'd by napi as `A | B`, which is precisely what our explicit `Node` definition is

Reply:
Sounds great. But I wonder if we should define/reuse this type here, instead of repeating it in all callsites:

```rust
pub type Node = Either<RuleNode, TokenNode>;
```

Reply:
Also, I wonder if this is a good use case for `impl Into<Node> for RustNode` instead of the custom `into_either()`.

Reply:
I would have expected this to update the generated `index.d.ts` in this PR. Not sure how the CI is passing without it :O

Reply:
As I mentioned in the OP, I deliberately did not remove the custom type; we still retain the `cst.Node` custom type and the explicit `ts_return_type` annotations for the DX.